File: app/src/main/java/com/apanda/base/Module/Gank/presenter/ItemPresenter.java
package com.apanda.base.Module.Gank.presenter;


import com.apanda.base.Module.Gank.bean.ItemsBean;
import com.apanda.base.Module.Gank.contract.ItemContract;
import com.apanda.base.base.baserx.RxSubscriber;

public class ItemPresenter extends ItemContract.Presenter {

    public void loadBean(String type) {
        mRxManage.add(mModel.loadCatoBean(type).subscribe(new RxSubscriber<ItemsBean>(mContext, false) {
            @Override
            public void onStart() {
                mView.showLoading("请稍后...");
                super.onStart();
            }

            @Override
            protected void _onNext(ItemsBean _gankBean) {
                mView.returnItemsBean(_gankBean);
                mView.stopLoading();
            }

            @Override
            protected void _onError(String message) {
                mView.showErrorTip(message);
            }
        }));
    }
}

File: app/src/main/java/com/apanda/base/Module/Gank/presenter/ItemPresenter.kt
package com.apanda.base.Module.Gank.presenter


import com.apanda.base.Module.Gank.bean.ItemsBean
import com.apanda.base.Module.Gank.contract.ItemContract
import com.apanda.base.base.baserx.RxSubscriber

class ItemPresenter : ItemContract.Presenter() {

    fun loadBean(type: String) {
        mRxManage.add(mModel.loadCatoBean(type).subscribe(object : RxSubscriber<ItemsBean>(mContext!!, false) {
            override fun onStart() {
                mView.showLoading("请稍后...")
                super.onStart()
            }

            override fun _onNext(_gankBean: ItemsBean) {
                mView.returnItemsBean(_gankBean)
                mView.stopLoading()
            }

            override fun _onError(message: String) {
                mView.showErrorTip(message)
            }
        }))
    }
}